.github/workflows/main.yml

```diff
@@ -0,0 +1,54 @@
+name: GitHub Actions
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: "Python ${{ matrix.python }}, Test: ${{ matrix.lang }}"
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python:
+          - 3.5
+          - 3.6
+          - 3.7
+          - 3.8
+        lang:
+          - "API"
+          - "c_lang or python or java or go_lang or javascript or php or haskell or ruby"
+          - "c_sharp or visual_basic or powershell"
+          - "r_lang or dart"
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 5
+      - name: Setup Python ${{ matrix.python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+      - name: Setup dependencies and run tests
+        run: |
+          if [[ "${{ matrix.lang }}" == "API" ]]; then
+              export TEST="API";
+              export COVERALLS_REPO_TOKEN="COVERALLS TOKEN FOR M2CGEN";
```

Comment:
FIXME

Reply:
I'm not quite sure how to overcome this. It doesn't seem to be a good idea to expose the API token and make it a part of the publicly available source code. Or am I missing something?

Reply:
Shall we use a Coveralls action (https://github.com/marketplace/actions/coveralls-github-action) instead of manually invoking the `coveralls` binary in the `test.sh` script?